lib/open_project/translations/models/locales_updater.rb

```diff
@@ -0,0 +1,189 @@
+    if dir['files'].find {|f| f['name'] == 'en.yml'}
+      crowdin.update_file [{dest: "/#{@crowdin_directory}/en.yml", source: path_to_translation.to_s, title: 'OpenProject wording', export_pattern: '%two_letters_code%.yml'}], type: 'yaml'
+    else
+      crowdin.add_file [{dest: "/#{@crowdin_directory}/en.yml", source: path_to_translation.to_s, title: 'OpenProject wording', export_pattern: '%two_letters_code%.yml'}], type: 'yaml'
+    end
+
+    # either add or update the english javascript translation file
+    path_to_translation = File.join 'config', 'locales', 'js-en.yml'
+    dir = crowdin.project_info['files'].find {|f| f['name'] == @crowdin_directory && f['node_type'] == 'directory'}
+    if dir['files'].find {|f| f['name'] == 'js-en.yml'}
+      crowdin.update_file [{dest: "/#{@crowdin_directory}/js-en.yml", source: path_to_translation.to_s, title: 'OpenProject JavaScript wording', export_pattern: 'js-%two_letters_code%.yml'}], type: 'yaml'
+    else
+      crowdin.add_file [{dest: "/#{@crowdin_directory}/js-en.yml", source: path_to_translation.to_s, title: 'OpenProject JavaScript wording', export_pattern: 'js-%two_letters_code%.yml'}], type: 'yaml'
+    end
+  end
+
+  def self.request_build
+    # todo what errors to catch? maybe this could run into a timeout
+    crowdin = create_crowdin_handle
+    crowdin.export_translations
+  end
+
+  def self.create_crowdin_handle
+    Crowdin::API.new project_id: @project_id, api_key: @api_key
+  end
+
+  def self.download_and_replace_locales
+    # todo what errors to catch here?
+    # todo delete all locales here? maybe for the case that
+    # we do not support a language anymore.
+    begin
+      languages_files = create_temp_file('crowdin_translations')
+      download_from_crowding(languages_files.path)
+
+      target_directory = Pathname(File.join('config', 'locales'))
+      Zip::File.open(languages_files.path) do |zip_file|
+        zip_file.glob("*/#{@crowdin_directory}/*.yml").each do |entry|
+          language_name = entry.name.split('/').first # the file is put in a directory containing the language name
+
+          # only take translations with enough percent translated
+          # todo do we require 100% here?
+          next unless translation_status_high_enough?(language_name, 100)
+
+          filepath = target_directory.join "#{js_translation?(Pathname.new(entry.name)) ? 'js-' : ''}#{language_name}.yml"
+
+          File.delete(filepath) if File.file?(filepath)
+          File.open(filepath, 'wb') do |file|
+            file.write entry.get_input_stream.read
+          end
+        end
+      end
+    ensure
+      unlink_temp_file(languages_files)
+    end
+  end
+
+  def self.translation_status_high_enough?(code, percent)
+    @translations_statuses ||= begin
+      crowdin = create_crowdin_handle
+      crowdin.translations_status
+    end
+    translation_status = @translations_statuses.select do |translation|
+      translation['code'] == code
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.